frontend/src/api/ApiProvider.ts

```diff
@@ -0,0 +1,41 @@
+import { 
+  BASE_API_URL,
+  LOCAL_STORAGE_API_KEY
+} from '@/helpers/constants';
+
+interface IHeaders {
+  [n: string]: string
+};
+
+export const createRequest = async (
+    endpoint: string, 
+    method: string, 
+    data: object | null = null
+  ) => {
+  const url = `${BASE_API_URL}/${endpoint}`;
+  const headers: IHeaders = {
+    'Content-Type': 'application/json'
+  };
+  
+  const token = localStorage.getItem(LOCAL_STORAGE_API_KEY);

```

Comment:
The auth token must be stored only in http-only cookies otherwise it will be a big security issue

Reply:
Changed the logic to working with cookies